token/program-2022/src/extension/memo_transfer/mod.rs

```diff
@@ -0,0 +1,36 @@
+use {
+    crate::{
+        extension::{Extension, ExtensionType, StateWithExtensionsMut},
+        state::Account,
+    },
+    bytemuck::{Pod, Zeroable},
+};
+
+/// Memo Transfer extension instructions
+pub mod instruction;
+
+/// Memo Transfer extension processor
+pub mod processor;
+
+/// Memo Transfer extension for Accounts
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
+pub struct MemoTransfer {
+    /// Require transfers into this account to be accompanied by a memo
+    pub require_incoming_transfer_memos: PodBool,
+}
+impl Extension for MemoTransfer {
+    const TYPE: ExtensionType = ExtensionType::MemoTransfer;
+}
+
+type PodBool = u8;
+
+/// Determine if a memo is required for transfers into this account
+pub fn memo_required(mut account_state: StateWithExtensionsMut<Account>) -> bool {
+    if let Ok(extension) = account_state.get_extension_mut::<MemoTransfer>() {
```

Comment:
nit: it might be worth adding a `get_extension` call to `StateWithExtensionsMut` to avoid having to take this by mut

Reply:
Done